### ISSUE TITLE
npm prune gets broken if you set depth in npm config

### ISSUE THREAD

**Issue by metalim** [npm/npm#12558]
```
┌(metalim@iMaxim)─(✓)─(2016-05-03 14:17:41)─(604Gi / 930Gi)
├(~/Sites/test_client)─(16 files, 280b)
└> npm ls --depth=0
test_client@1.0.0 /Users/metalim/Sites/test_client
├── browser-sync@2.12.5
├── browserify@13.0.0
├── coffee-script@1.10.0
├── coffeeify@2.0.1
├── cson@3.0.2
├── fastclick@1.0.6
├── gulp@4.0.0-alpha.2 (git://github.com/gulpjs/gulp.git#b4d89c391b40cd29e8526e0391981eef67cfccab)
├── gulp-git@1.7.1
├── gulp-uglify@1.5.3
├── gulp-util@3.0.7
├── howler@2.0.0-beta12 (git://github.com/goldfire/howler.js.git#db831e03cf696d1ca5cfdd8f87d66e66da2c5029)
├── minimatch@3.0.0
├── nouislider@8.5.1
├── parcelify@2.1.0
├── q@1.4.1
├── vinyl-buffer@1.0.0
├── vinyl-source-stream@1.1.0
└── watchify@3.7.0

┌(metalim@iMaxim)─(✓)─(2016-05-03 14:25:02)─(604Gi / 930Gi)
├(~/Sites/test_client)─(16 files, 280b)
└> npm prune

```

This works as expected.

Now, here's the surprise in npm v3+ (works fine in npm v2.*):

```
┌(metalim@iMaxim)─(✓)─(2016-05-03 14:25:08)─(604Gi / 930Gi)
├(~/Sites/test_client)─(16 files, 280b)
└> npm config set depth 0

┌(metalim@iMaxim)─(✓)─(2016-05-03 14:25:14)─(604Gi / 930Gi)
├(~/Sites/test_client)─(16 files, 280b)
└> npm prune
unbuild align-text@0.1.4
unbuild center-align@0.1.3
unbuild archy@1.0.0
unbuild findup-sync@0.3.0
unbuild flagged-respawn@0.3.2
unbuild formidable@1.0.17
unbuild interpret@1.0.1
unbuild lazy-cache@1.0.4
unbuild lodash.sortby@4.4.2
unbuild lodash.isarray@3.0.4
unbuild longest@1.0.1
unbuild lru-cache@2.7.3
unbuild matchdep@1.0.1
unbuild mute-stdout@1.0.0
unbuild os-homedir@1.0.1
unbuild pretty-hrtime@1.0.2
unbuild rechoir@0.6.2
unbuild semver-greatest-satisfied-range@1.0.0
unbuild sigmund@1.0.1
unbuild stack-trace@0.0.9
unbuild semver-regex@1.0.0
unbuild right-align@0.1.3
unbuild tildify@1.2.0
unbuild wreck@6.3.0
unbuild v8flags@2.0.11
unbuild wordwrap@0.0.2
unbuild lodash._getnative@3.9.1
unbuild user-home@1.1.1
unbuild liftoff@2.2.1
unbuild lodash.isarguments@3.0.8

```

**Comment by KenanY** [comment]
Looks like `depth` has affected `npm prune` since at least as far back as 2012. I'm not sure if this behavior is intentional or not, but it's definitely [not documented](https://docs.npmjs.com/cli/prune). Will need one of the core team members to clear this up :)

**Comment by KenanY** [comment]
Maybe related/same-as: https://github.com/npm/npm/issues/10191

**Comment by Moeriki** [comment]
AFAICS it was introduced in commit 660caf6, and first released with tag `1.0.0rc3`.
